src/SSHLibrary/pythonclient.py

```diff
@@ -95,10 +95,19 @@ def _login(self, username, password, allow_agent=False, look_for_keys=False):
         except paramiko.AuthenticationException:
             raise SSHClientException
 
-    def _login_with_public_key(self, username, key_file, password, allow_agent, look_for_keys):
+    def _login_with_public_key(self, username, key_file, password, allow_agent, look_for_keys, proxy_cmd=None):
         self.config.host = self._read_ssh_config_host(self.config.host)
         try:
-            self.client.connect(self.config.host, self.config.port, username,
+            if proxy_cmd:
```

Comment:
The if branch isn't necessary because the default value of `sock` is `None` and will be passed as `None` if no `proxy_cmd` is given. Just adding `sock=proxy` argument to `connect` call is sufficient on this method.